### PR TITLE
 boot: set extra kernel command line arguments when making a recovery system bootable

### PR DESCRIPTION
**Pull request by bboozzoo**
The branch sets up kernel command line arguments for the seed system, which makes them appear in install & recovery modes. The PR includes a simple spread test to verify that recovery mode has the right command line arguments.

Based on top of #10120.

### CODE REVIEW

File: tests/nested/manual/core20-custom-kernel-commandline/task.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    "$TESTSTOOLS"/nested-state create-vm core
+
+execute: |
+    #shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB/nested.sh"
+
+    # TODO: verify /proc/cmdline has the right arguments when run system support
+    # lands
+
+    echo "Transition to recover mode to verify kernel command line"
```
Comment (mvo5) [inline tests/nested/manual/core20-custom-kernel-commandline/task.yaml:35]: Thanks for adding this test

**Review by mvo5 — APPROVED** [review]
Thanks, looks fine!

**Review by pedronis — APPROVED** [review]
thank you

**Review by anonymouse64 — COMMENTED** [review]
post merge LGTM, thanks for this

**Pull request merged**